frame/mosaic-vault/src/lib.rs

```diff
@@ -0,0 +1,904 @@
+
+			T::Currency::mint_into(asset_id, &to, withdrawable_balance).map_err(|_|Error::<T>::MintToFailed)?;
+
+			Self::decrease_total_value_transferred(asset_id, withdrawable_balance)?;
+             
+		    Self::deposit_event(Event::LiquidityMoved {sender, to, withdrawable_balance});
+
+			Ok(().into())
+		}
+
+		#[pallet::weight(10_000)]
+		pub fn pause(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+
+			 T::AdminOrigin::ensure_origin(origin.clone())?;
+
+			let sender = ensure_signed(origin)?;
+
+			ensure!(Self::pause_status() == false, Error::<T>::ContractPaused);
+			 <PauseStatus<T>>::put(true);
+			 Self::deposit_event(Event::Pause{sender});
+
+			 Ok(().into())
+		}
+
+		#[pallet::weight(10_000)]
+		pub fn un_pause(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+
+			T::AdminOrigin::ensure_origin(origin.clone())?;
+
+			let sender = ensure_signed(origin)?;
+
+			 <PauseStatus<T>>::put(false);
+			 Self::deposit_event(Event::UnPause{sender});
+
+			 Ok(().into())
+		}
+ 	}
+
+	impl<T: Config> Pallet<T> {
+		fn account_id() -> T::AccountId {
+			T::PalletId::get().into_account()
+		}
+
+		fn get_fee_address() -> T::AccountId {
+			T::FeeAddress::get()
+		}
+
+		fn increment_nonce() -> T::Nonce {
+
+			let mut nonce = Self::nonce();
+
+			nonce += 1u8.into();
+
+			<Nonce<T>>::put(nonce);
+
+			nonce
+		}
+
+		fn get_current_token_liquidity(asset_id: T::AssetId) -> Result<T::Balance, DispatchError> {
+		
+			let available_funds = Self::total_value_transferred(asset_id);
+
+			let liquidity = available_funds.checked_sub(&Self::in_transfer_funds(asset_id)).ok_or(Error::<T>::Underflow)?;
+
+			 Ok(liquidity)
+		}
+
+		fn only_supported_remote_token(remote_network_id: T::RemoteNetworkId, asset_id:T::AssetId) -> Result<T::RemoteAssetId, DispatchError> {
+			
+			let remote_asset_id = <RemoteAssetId<T>>::try_get(remote_network_id, asset_id).map_err(|_|Error::<T>::UnsupportedToken)?;
+
+			Ok(remote_asset_id)
+		}
+
+		fn generate_deposit_id(
+			remote_network_id: T::RemoteNetworkId,
+			destination_address: &T::AccountId,
+			pallet_account_id: T::AccountId,
+		) -> [u8; 32] {
+
+			let mut encoded_remote_network_id = Encode::encode(&remote_network_id);
+
+			let mut encoded_block_number = Encode::encode(&<frame_system::Pallet<T>>::block_number());
+
+            let mut encoded_destination_address = Encode::encode(&destination_address);
+
+			let mut encoded_pallet_account_id = Encode::encode(&pallet_account_id);
+
+			let mut encoded_nonce = Encode::encode(&Self::increment_nonce());
+
+			let mut encoded_data = Vec::new();
+			encoded_data.append(& mut encoded_remote_network_id);
+			encoded_data.append(& mut encoded_block_number);
+			encoded_data.append(& mut encoded_destination_address);
+			encoded_data.append(& mut encoded_pallet_account_id);
+			encoded_data.append(& mut encoded_nonce);
+
+			let deposit_id = keccak_256(&encoded_data);
```

Comment:
vec![
    remote_network_id.encode(),
    ..
    Self::increment_nonce().encode(),
]

Reply:
keccak_256 wants a &[u8] , .append moves individual elements into new array, however using vet![] with encode results in Vec<Vec<u8>>